prusti-viper/src/encoder/pure_function_encoder.rs

```diff
@@ -882,6 +882,22 @@ impl<'p, 'v: 'p, 'tcx: 'v> BackwardMirInterpreter<'tcx>
                                 state
                             }
 
+                            "std::cmp::PartialEq::eq"
+                            | "std::cmp::PartialEq::ne" => {
+                                // we only get here if `has_structural_eq_impl` returned false
+                                // above
```

Comment:
Can we keep this in the same match arm but instead of an `if` guard, we return the error `if !has_structural_eq_impl...`?

Reply:
sounds good, refactored to have all in the same match case.